src/main/scala/esmeta/lang/util/Stringifier.scala

```diff
@@ -197,7 +197,10 @@ class Stringifier(detail: Boolean, location: Boolean) {
       case LengthExpression(expr) =>
         app >> "the length of " >> expr
       case SubstringExpression(expr, from, to) =>
-        app >> "the substring of " >> expr >> " from " >> from >> " to " >> to
+        app >> "the substring of " >> expr >> " from " >> from
+        to match
```

Comment:
Try to use the `map` function of the `Option[_]` type, or [For Comprehension](https://docs.scala-lang.org/tour/for-comprehensions.html). Do not use curly braces if possible.